src/firmware/app/control/trajectory_planner_test.cpp

```diff
@@ -0,0 +1,529 @@
+extern "C"
+{
+#include "firmware/app/control/trajectory_planner.h"
+}
+
+#include <gtest/gtest.h>
+#include <math.h>
+
+#include <vector>
+
+#include "firmware/shared/math/polynomial_1d.h"
+#include "firmware/shared/math/polynomial_2d.h"
+#include "firmware/shared/math/vector_2d.h"
+
+class TrajectoryPlannerTest : public testing::Test
+{
+   protected:
+    virtual void SetUp() {}
+
+    virtual void TearDown() {}
+
+    static std::vector<float> getSpeedsFromTrajectory(Trajectory_t* trajectory,
+                                                      float inital_speed)
+    {
+        std::vector<float> velocity;
+
+        // Calculate the delta-position and the delta-time
+        velocity.push_back(inital_speed);
+
+        for (unsigned i = 1; i < trajectory->num_elements - 1; i++)
+        {
+            const float dx = trajectory->trajectory_elements[i + 1].position.x -
+                             trajectory->trajectory_elements[i].position.x;
+            const float dy = trajectory->trajectory_elements[i + 1].position.y -
+                             trajectory->trajectory_elements[i].position.y;
+
+            const float dt = trajectory->trajectory_elements[i + 1].time -
+                             trajectory->trajectory_elements[i].time;
+
+            velocity.push_back(sqrt((pow(dx, 2) + pow(dy, 2))) / dt);
+        }
+        return velocity;
+    }
+
+    static std::vector<float> getAccelerationsFromSpeed(std::vector<float> speeds,
+                                                        Trajectory_t* trajectory)
+    {
+        std::vector<float> acceleration;
+
+        // Calculate the delta-position and the delta-time
+        acceleration.push_back(0.0);
+
+        for (unsigned i = 1; i < trajectory->num_elements - 2; i++)
+        {
+            const float dv = speeds[i + 1] - speeds[i];
+            const float dt = trajectory->trajectory_elements[i + 1].time -
+                             trajectory->trajectory_elements[i].time;
+
+            acceleration.push_back(dv / dt);
+        }
+        return acceleration;
+    }
+};
+
+TEST_F(TrajectoryPlannerTest, check_trajectory_length_and_end_points_match_path)
```

Comment:
Because w're doing a bit of work here to generate the "expected" values, could we break this into 2 separate tests so they're a bit smaller and it's more obvious what goes with what.

Reply:
Broke up into 2 unit tests. One for path length and one for endpoints